### PR TITLE
externalconn: encrypt,decrpyt when creating KMS external connection

### PR DESCRIPTION
**Pull request by adityamaru**
This patch does a couple of things:

`CREATE EXTERNAL CONNECTION` for gcp KMS now recognizes the `gcp`
scheme instead of the `gs` scheme. This is motivated by a future change
where we will add support for google ExternalStorage which is also
represented by the `gs` scheme today. Since we cannot have a collision
of schemes and this is new functionality we are okay asking users to
use `gcp` when referring to a KMS URI, and `gs` when referring to a
storage URI. For uniformity, the existing raw URI google KMS will also
start recognizing `gcp` in addition to the existing `gs` scheme.

This patch also adds a step in the creation of a KMS external connection
that encrypts and decrypts some sentinel content to validate that the resource
is reachable and correctly configured.

Release note (sql change): Google Cloud KMS will now accept the `gcp`
scheme alongwith the existing `gs` scheme. External Connections will only
recognize the `gcp` scheme when being created to represent a KMS resource.

### CODE REVIEW

File: pkg/cloud/gcp/gcs_kms.go [hunk 1]
```diff
@@ -24,7 +24,10 @@ import (
 	"google.golang.org/protobuf/types/known/wrapperspb"
 )
 
-const gcsScheme = "gs"
+const (
```
Comment (rhu713) [inline pkg/cloud/gcp/gcs_kms.go:27]: should we rename this file to `gcp_kms.go`?

**Review by rhu713 — APPROVED** [review]

**Review by benbardin — COMMENTED** [review]
I don't love the prefix `gcp` to signify a kms...perhaps I'm missing something? But it's not at all obvious to me that while `gs` represents google storage, `gcp` represents not "google cloud platform," but actually "the specific component of google cloud platform that handles keys." This is compounded by external storage accepting both prefixes...

Could we instead invent a different scheme for keys? Perhaps `gkms`?

File: pkg/cloud/gcp/gcs_kms.go (hunk 1 above)
Comment (adityamaru) [reply pkg/cloud/gcp/gcs_kms.go:27]: done!

**Comment by adityamaru** [comment]
@benbardin reworked to now accept `gcp-kms` as the scheme, check it out!

**Review by benbardin — APPROVED** [review]

**Comment by adityamaru** [comment]
Failures are unrelated!

bors r=benbardin

**Pull request merged**